src/MTProto.php

```diff
@@ -920,6 +925,9 @@ public function wakeup(SettingsAbstract $settings, APIWrapper $wrapper): void
         $this->initPromise = $deferred->getFuture();
 
         try {
+            // Update settings from constructor
```

Comment:
But this is the point. We initialize DdArray in cleanupProperties with wrong settings.

Reply:
What do you suggest? Without this fix you litterally cant dissable any database in settings, because they initialize in cleanup.

Reply:
This will break database migration.

If you first set the database type/name to mysql/"MadelineProto_1", wait for the migration from memory, then stop madelineproto, and pass a new database name postgres/"MadelineProto_2" to the constructor when starting madelineproto, a "Could not extract session from database!" error will be emitted, because extraction from the new database will be attempted, instead of loading data from the old one and migrating to the new one.

Reply:
Then we need to refactor or fix migrations? Because it is wrong to initialize before correct settings are set.

Reply:
What's the specific usecase you're worried about? Settings are generally not that critical to require an immediate update.

Reply:
As i said before - i tried to disable minDatabase, and i cant. Because its initialized before settings update. Same with other databases.

Reply:
Im getting alot of flood_waits and MSG_INVALID errors from minDataBase refresh and its usless for my purposes, i dont work with participants.

Reply:
Min database refreshing was removed in the latest beta

Reply:
Also it looks like this bug also prevent other DB settings change, like: host, port, username...

Reply:
If we want migrations we can serialize copy of old settings in Db class, as it was in initial version of DB. Or we can keep diff after settings update and check it in DB.